src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -457,6 +474,45 @@ def _get_managed_db_resource_id(cli_ctx, resource_group_name, managed_instance_n
         quote(database_name))
 
 
+def _to_filetimeutc(dateTime):
+    NET_epoch = datetime(1601, 1, 1)
+    UNIX_epoch = datetime(1970, 1, 1)
+
+    epoch_delta = (UNIX_epoch - NET_epoch)
+
+    log_time = parse(dateTime)
+
+    net_ts = calendar.timegm((log_time + epoch_delta).timetuple())
+
+    # units of seconds since NET epoch
+    filetime_utc_ts = net_ts * (10**7) + log_time.microsecond * 10
+
+    return filetime_utc_ts
+
+
+def _get_managed_dropped_db_resource_id(
+        cli_ctx,
+        resource_group_name,
+        managed_instance_name,
+        database_name,
+        deletion_date):
+    '''
+    Gets the Managed db resource id in this Azure environment.
+    '''
+
+    # url parse package has different names in Python 2 and 3. 'six' package works cross-version.
+    from six.moves.urllib.parse import quote  # pylint: disable=import-error
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    return ('/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/managedInstances/'
```

Comment:
You can consider using resource_id in msrestazure. `from msrestazure.tools import resource_id`